README.md

```diff
@@ -55,6 +75,13 @@ The width (in pixels) that the image needs to be displayed in the flow.  The hei
 ## Image
 Specify how the input image will be delivered to this node.  By default the image needs to be delivered in the ```msg.payload``` of the input message.
 
+But there are a number of locations where this node can get the input image:
+1. ***msg***: image will be read from an input message.
+1. ***flow***: image will be read from in a flow variable.
+1. ***global***: image will be read from a global variable.
+1. ***string***: image will be specified as a fixed base64 string.
```

Comment:
Can't you auto detect string or buffer on the input type ?

Reply:
Yes we already do that, when the option is '_msg_'.  Suppose you specify in the config screen that the image will arrive via ```msg.payload```, then we detect whether that image (inside the payload) is a string, buffer or jimp object.  

But in this case the 'str' and 'bin' options allow an image to be hardcoded:
![image](https://user-images.githubusercontent.com/14224149/71605277-edd4e600-2b67-11ea-9b84-9f223ccba789.png)
Does this not makes sense to you?

Reply:
errr - it does now you explained it... - but I don't "get it" as part of the purpose of this node... I thought it was supposed to be like the debug node so you could see a image that was passed in - If you are just fixing it to be a static image that is configured in the dialog then there is no point in displaying it in the editor as the editor isn't a dashboard :-)

Reply:
Damn, this time you got me ;-)
Then indeed the flow editor is becoming a dashboard.  No arguments against it, and it doesn't even hurt :-)
So we only keep 'msg' as an option.   Will change the code ...

Reply:
When 'msg' becomes the only option then:
+ ```node.dataType``` will always contain "_msg_"
+ ```node.data``` will contain the fieldname e.g. "_payload_"

Currently I use this to get the image:
```
node.on("input", function(msg) {
    RED.util.evaluateNodeProperty(node.data, node.dataType, node, msg, (err, image) => {
       ...
    }
}
```
But since we are only dealing with '_msg_', won't you think it would be faster (I mean better performance) when I simply do this:
```
function ImageNode(config) {
    RED.nodes.createNode(this, config);
    this.data = config.data || "payload";
    ..

    node.on("input", function(msg) {
        image = msg[node.data]
        ...
    }
}}
```
Moreover we will have 1 asynchronous callback less, which makes the code again easier to read ...  In the above code I have also added "_payload_" as default value, when no field is specified.  Simply because it is a rare case (since it is a mandatory field on the config screen)...

Reply:
Feel free to tidy it up again.

Reply:
actually no - if node.data was  "payload.foo.bar" - then msg[payload.foo.bar] would not evaluate correctly (should be expanded to  msg[payload][foo][bar] instead - so no leave it)